### PR TITLE
Report error status

### PR DESCRIPTION
**Pull request by davidharrigan**
This PR updates the error object so that helpful message is returned and also someone using this library can make decisions based on the error message or the status code. 

Also added `EnableSSLAlert` option to `Tests`

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -98,24 +98,25 @@ func (c *Client) doRequest(r *http.Request) (*http.Response, error) {
...
-			statusCode: resp.StatusCode,
-		}
+		return nil, NewHTTPError(resp)
 	}
 
 	var aer autheticationErrorResponse
 
 	// We read and save the response body so that if we don't have error messages
 	// we can set it again for future usage
+	defer resp.Body.Close()
```
Comment (jelmersnoeck) [inline client.go]: Why did you move it down here? We'd still want it to be handled before the previous conditional. If the status code is anything else, we now won't close the body.

**Review by jelmersnoeck — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (davidharrigan) [reply client.go]: woops - good catch.  I meant to add `resp.Body.Close()` to `NewHTTPError`.  I moved this down here so the function won't try to close the Body twice.  I moved it down to one line above where we actually try to interact with the Body in the function.

Comment (davidharrigan) [reply client.go]: I decided to just move it back and not add it to NewHTTPError so it's done in one place.

**Review by jelmersnoeck — APPROVED** [review]
